MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/ImageSourceActivity.java

```diff
@@ -0,0 +1,134 @@
+  private Handler handler;
+  private Runnable runnable;
+  private static final String ID_IMAGE_SOURCE = "animated_image_source";
+  private static final String ID_IMAGE_LAYER = "animated_image_layer";
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_image_source);
+
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    // Add source
+    LatLngQuad quad = new LatLngQuad(
+      new LatLng(46.437, -80.425),
+      new LatLng(46.437, -71.516),
+      new LatLng(37.936, -71.516),
+      new LatLng(37.936, -80.425));
+    mapboxMap.addSource(new ImageSource(ID_IMAGE_SOURCE, quad, R.drawable.southeast_radar_0));
+
+    // Add layer
+    RasterLayer layer = new RasterLayer(ID_IMAGE_LAYER, ID_IMAGE_SOURCE);
+    mapboxMap.addLayer(layer);
+
+    // Loop the GeoJSON refreshing
+    handler = new Handler();
+    runnable = new RefreshImageRunnable(mapboxMap, handler);
+    handler.postDelayed(runnable, 100);
+  }
+
+  private static class RefreshImageRunnable implements Runnable {
+    private MapboxMap mapboxMap;
+    private Handler handler;
+    private int[] drawables;
+    private int drawableIndex;
+
+    RefreshImageRunnable(MapboxMap mapboxMap, Handler handler) {
+      this.mapboxMap = mapboxMap;
+      this.handler = handler;
+      drawables = new int[4];
+      drawables[0] = R.drawable.southeast_radar_0;
+      drawables[1] = R.drawable.southeast_radar_1;
+      drawables[2] = R.drawable.southeast_radar_2;
+      drawables[3] = R.drawable.southeast_radar_3;
+      drawableIndex = 1;
+    }
+
+    @Override
+    public void run() {
+      ((ImageSource) mapboxMap.getSource(ID_IMAGE_SOURCE)).setImage(drawables[drawableIndex++]);
+      if (drawableIndex > 3) {
+        drawableIndex = 0;
+      }
+      handler.postDelayed(this, 1000);
+    }
+  }
+
+  // Add the mapView lifecycle to the activity's lifecycle methods
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
```

Comment:
to avoid a crash on orientation change in tablets we should add:
    if (handler != null && runnable != null) {
          handler.removeCallbacks(runnable);
    }

Reply:
Does the trick on the tablet emulator. No crashing. Thanks.

![ezgif com-resize](https://user-images.githubusercontent.com/4394910/33636607-c10b3780-d9d1-11e7-8ad0-906fff11178c.gif)


Think we're good to go on this pr